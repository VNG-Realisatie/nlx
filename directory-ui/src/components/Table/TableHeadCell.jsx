import { node, oneOf } from 'prop-types'
import styled from 'styled-components'

const TableHeadCell = styled.th`
    display: table-cell;
    border-bottom: 1px solid #EAEAEA;
    padding: .65rem;

    font-size: .85rem;
    line-height: 1.25rem;
    font-weight: 700;
    color: #ADB3C6;
    overflow: hidden;
    text-overflow: ellipsis;
    text-transform: uppercase;
    white-space: nowrap;
    
    text-align: ${ p => p.align};

    &:last-child {
        padding-right: 1.5rem;
    }
`

TableHeadCell.propTypes = {
  children: node,
  align: oneOf(['left', 'center', 'right']),
}

TableHeadCell.defaultProps = {
  align: 'left',
}

export default TableHeadCell
