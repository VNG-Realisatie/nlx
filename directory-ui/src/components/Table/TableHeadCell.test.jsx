import React from 'react'
import { shallow, mount } from 'enzyme'
import TableHeadCell from './TableHeadCell'

it('should match snapshot', () => {
  expect(shallow(<TableHeadCell/>)).toMatchSnapshot()
})

it('should render the contents', () => {
  const wrapper = mount(<TableHeadCell>Heading</TableHeadCell>)
  expect(wrapper.text()).toEqual('Heading')
})

describe('text alignment', () => {
  it('should use left alignment as default', () => {
    const wrapper = shallow(<tbody><TableHeadCell/></tbody>)
    expect(wrapper.find(TableHeadCell).prop('align')).toEqual('left')
  })

  it('should support center alignment', () => {
    const wrapper = shallow(<tbody><TableHeadCell align="center"/></tbody>)
    expect(wrapper.find(TableHeadCell).prop('align')).toEqual('center')
  })

  it('should support right alignment', () => {
    const wrapper = shallow(<tbody><TableHeadCell align="right"/></tbody>)
    expect(wrapper.find(TableHeadCell).prop('align')).toEqual('right')
  })
})
