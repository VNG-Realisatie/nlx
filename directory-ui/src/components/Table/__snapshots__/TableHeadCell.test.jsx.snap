// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<StyledComponent
  align="left"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    display: table-cell;
    border-bottom: 1px solid #EAEAEA;
    padding: .65rem;

    font-size: .85rem;
    line-height: 1.25rem;
    font-weight: 700;
    color: #ADB3C6;
    overflow: hidden;
    text-overflow: ellipsis;
    text-transform: uppercase;
    white-space: nowrap;
    
    text-align: ",
          [Function],
          ";

    &:last-child {
        padding-right: 1.5rem;
    }
",
        ],
      },
      "defaultProps": Object {
        "align": "left",
      },
      "displayName": "styled.th",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "align": [Function],
        "children": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "th",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
